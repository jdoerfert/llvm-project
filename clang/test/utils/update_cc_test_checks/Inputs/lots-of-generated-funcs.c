// Check that the CHECK lines are generated for clang-generated functions
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fopenmp %s -emit-llvm -o - | FileCheck --check-prefix=OMP %s
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu %s -emit-llvm -o - | FileCheck --check-prefix=NOOMP %s

void t0() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}

void t1() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}

void t2() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}
