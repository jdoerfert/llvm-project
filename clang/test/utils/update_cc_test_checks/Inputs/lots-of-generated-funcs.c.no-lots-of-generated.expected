// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
// Check that the CHECK lines are generated for clang-generated functions
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fopenmp %s -emit-llvm -o - | FileCheck --check-prefix=OMP %s
// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu %s -emit-llvm -o - | FileCheck --check-prefix=NOOMP %s

// OMP-LABEL: @t0(
// OMP-NEXT:  entry:
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1:[0-9]+]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined. to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..1 to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..2 to void (i32*, i32*, ...)*))
// OMP-NEXT:    ret void
//
// NOOMP-LABEL: @t0(
// NOOMP-NEXT:  entry:
// NOOMP-NEXT:    ret void
//
void t0() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}

// OMP-LABEL: @t1(
// OMP-NEXT:  entry:
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..3 to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..4 to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..5 to void (i32*, i32*, ...)*))
// OMP-NEXT:    ret void
//
// NOOMP-LABEL: @t1(
// NOOMP-NEXT:  entry:
// NOOMP-NEXT:    ret void
//
void t1() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}

// OMP-LABEL: @t2(
// OMP-NEXT:  entry:
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..6 to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..7 to void (i32*, i32*, ...)*))
// OMP-NEXT:    call void (%struct.ident_t*, i32, void (i32*, i32*, ...)*, ...) @__kmpc_fork_call(%struct.ident_t* @[[GLOB1]], i32 0, void (i32*, i32*, ...)* bitcast (void (i32*, i32*)* @.omp_outlined..8 to void (i32*, i32*, ...)*))
// OMP-NEXT:    ret void
//
// NOOMP-LABEL: @t2(
// NOOMP-NEXT:  entry:
// NOOMP-NEXT:    ret void
//
void t2() {
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
  #pragma omp parallel
  {
  }
}
